backend/src/GraphQl/OpenIdConnect/Application/ApplicationMutations.cs

```diff
@@ -0,0 +1,259 @@
+
+        return new UpdateApplicationPayload(application);
+    }
+
+    [UseUserManager]
+    [Authorize(Policy = AuthConfiguration.WritePolicy)]
+    public async Task<DeleteApplicationPayload> DeleteApplicationAsync(
+        DeleteApplicationInput input,
+        OpenIddictApplicationManager<OpenIdApplication> applicationManager,
+        ClaimsPrincipal claimsPrincipal,
+        UserManager<User> userManager,
+        ApplicationDbContext context,
+        CancellationToken cancellationToken
+    )
+    {
+        if (!await OpenIdConnectAuthorization.IsAuthorizedToManageApplication(
+                input.ApplicationId,
+                applicationManager,
+                claimsPrincipal,
+                userManager,
+                context,
+                cancellationToken
+            ).ConfigureAwait(false))
+        {
+            return new DeleteApplicationPayload(
+                new DeleteApplicationError(
+                    DeleteApplicationErrorCode.UNAUTHORIZED,
+                    "You are not authorized to delete the application.",
+                    new[] { nameof(input), nameof(input.ApplicationId).FirstCharToLower() }
+                )
+            );
+        }
+        if (input.ApplicationId == Guid.Empty)
+        {
+            return new DeleteApplicationPayload(
+                new DeleteApplicationError(DeleteApplicationErrorCode.UNKNOWN,
+                "Empty Application Id",
+                Array.Empty<string>()));
+        }
+
+        var application = await applicationManager.FindByIdAsync(input.ApplicationId.ToString(), cancellationToken).ConfigureAwait(false);
+
+        if (application is null)
+        {
+            return new DeleteApplicationPayload(
+                new DeleteApplicationError(
+                    DeleteApplicationErrorCode.UNKNOWN_APPLICATION,
+                    "Unknown application.",
+                    new[] { nameof(input), nameof(input.ApplicationId).FirstCharToLower() }
+                )
+            );
+        }
+
+        await applicationManager.DeleteAsync(application, cancellationToken).ConfigureAwait(false);
+
+        DeleteApplicationInstitutionConnection(application, context, cancellationToken);
+
+        return new DeleteApplicationPayload();
+    }
+
+    private static void UpdateApplicationDescriptor(UpdateApplicationInput input, OpenIddictApplicationDescriptor descriptor)
+    {
+        descriptor.ClientId = input.ClientId;
+        descriptor.DisplayName = input.DisplayName;
+        descriptor.RedirectUris.Clear();
+        descriptor.PostLogoutRedirectUris.Clear();
+        descriptor.RedirectUris.Add(new Uri(input.RedirectUri, UriKind.Absolute));
+        descriptor.PostLogoutRedirectUris.Add(new Uri(input.PostLogoutRedirectUri, UriKind.Absolute));
+        descriptor.Permissions.RemoveWhere(permission => permission.Contains(AuthConfiguration.ScopePrefixApi));
```

Comment:
`ScopePrefixApi` is currently only the string `"api"`. I would add the colon `":"` to this prefix to make its usage here like this less error-prone.